pkg/client/gcp/types.go

```diff
@@ -50,9 +50,13 @@ func (c *Client) ListKubernetesFirewallRulesForNetwork(ctx context.Context, proj
 // ListKubernetesRoutesForNetwork returns a list of all routes within the shoot network which have the namespace as prefix.
 func (c *Client) ListKubernetesRoutesForNetwork(ctx context.Context, project, network, namespace string) ([]string, error) {
 	var routes []string
+	// Gce cloudprovider implementation allows only cluster names up to 26 chars, rest is trimmed.
+	if len(namespace) > 26 {
```

Comment:
Wouldn't it be safer to just check for the prefix `shoot--`? Maybe GCP changes this limit (26 chars) to something else in the future, then we have to adapt the code?

Reply:
It wouldn't be safer, because we skip the project and name check before route deletion but we would be independent of the cloud provider implementation.

Reply:
Which way shall we go?

Reply:
My feeling is that checking for `shoot--` prefix is more sustainable than checking for length 26. In the end both will work. I'm fine with both. Just tell me how you will proceed.

Reply:
Then let's go the more sustainable way!